pallets/identity/src/lib.rs

```diff
@@ -259,25 +251,35 @@ decl_module! {
             };
             use polymesh_common_utilities::traits::identity::runtime_upgrade::LinkedKeyInfo;
 
-            migrate_map::<LinkedKeyInfo, _>(
-                b"identity",
-                b"KeyToIdentityIds",
-                |_| Empty
-            );
+            let spec_version = frame_system::LastRuntimeUpgrade::get().map_or(0, |upgrade| upgrade.spec_version.0);
+
+            if spec_version == 2000
+            {
+                migrate_map::<LinkedKeyInfo, _>(
+                    b"identity",
+                    b"KeyToIdentityIds",
+                    |_| Empty
+                    );
             // Migrate secondary key permissions to the new type
             migrate_map::<IdentityWithRolesOld<T::AccountId>, _>(
-                b"identity",
-                b"DidRecords",
-                |_| Empty
-            );
+                    b"identity",
+                    b"DidRecords",
+                    |_| Empty
+                    );
             // Remove roles from Identities
             StorageIterator::<IdentityWithRoles<T::AccountId>>::new(b"identity", b"DidRecords")
-                .drain()
-                .map(|(key, old)|  (key, DidRecord {
-                    primary_key: old.primary_key,
-                    secondary_keys: old.secondary_keys,
-                }))
+                    .drain()
+                    .map(|(key, old)|  (key, DidRecord {
+                        primary_key: old.primary_key,
+                        secondary_keys: old.secondary_keys,
+                    }))
                 .for_each(|(key, new)| put_storage_value(b"identity", b"DidRecords", &key, new));
+            }
+
+            if spec_version == 2001 {
+                // Migrate claims
+                <Claims>::translate(migration::migrate_claim);
```

Comment:
Instead of defining custom logic, we can reuse our `migrate_map` and migrate derive macros for consistency.

Reply:
That is the substrate [IterableStorageDoubleMap::translate](https://substrate.dev/rustdocs/v2.0.0/frame_support/storage/trait.IterableStorageDoubleMap.html#tymethod.translate).
In that case it seems a better choice in performance terms, because it does not need to re-calculate the hash of the new key (new_key == old_key, using only `unhashed::put`)